File: server/src/protocol/BlindsStatusResponseBuilder.cpp
#include "BlindsStatusResponse.h"
#include "BlindsStatusResponseBuilder.h"

using namespace std;
using namespace communication;

BlindsStatusResponseBuilder::BlindsStatusResponseBuilder()
{}

BlindsStatusResponseBuilder::~BlindsStatusResponseBuilder()
{}

unique_ptr<Response> BlindsStatusResponseBuilder::create(const vector<uint8_t> &dataInBytes)
{
    const auto status = static_cast<BlindsStatus>(dataInBytes[INIT_DATA_POSITION]);
    const auto mode = static_cast<BlindsMode >(dataInBytes[INIT_DATA_POSITION+1]);

    auto command = make_unique<BlindsStatusResponse>(status, mode);

    return move(command);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsStatusResponseBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class BlindsStatusResponseBuilder implements ResponseBuilder, Closeable
	{
		public BlindsStatusResponseBuilder()
		{
		}
		public final void close()
		{
		}

        @Override
        public Response create(ArrayList<Integer> commandInBytes)
        {
            BlindsStatus status = BlindsStatus.forValue(commandInBytes.get(initialDataPosition));
            BlindsMode mode = BlindsMode.forValue(commandInBytes.get(initialDataPosition+1));

            Response response  = new BlindsStatusResponse(status, mode);
            return response;
        }
	}